### PR TITLE
fix(triage): trim noisy zero counters and shorten thread bullets

### PR DESCRIPTION
**Pull request by esraagamal6**
## Summary
- Real nightly Slack output showed `:wrench: infrastructure: 0` and `:game_die: flakiness: 0` cluttering the summary line — now only non-zero category counters are shown.
- Thread bullets repeated the full test title and verbose per-link filler text — now compacted to one operationId/expected/actual line plus a single combined links line.

Follow-up to #487 (merged) — that PR had already landed before this formatting fix was ready, so this is a separate PR off current `main` rather than a reopened #487.

## Test plan
- [x] Re-ran the full synthetic fixture battery (8 cases covering every medic-ping/category/link-type edge case) through both `summary` and `thread` modes — no change to trigger logic, only formatting
- [x] Verified against the exact real nightly scenario the user reported (`createVersion` failure) — output matches the selected compact format
- [x] `shellcheck` clean
- [x] `bash -n` syntax check clean

### CODE REVIEW

File: scripts/triage/hub-triage-format-slack.sh [hunk 1]
```diff
@@ -203,21 +217,26 @@ case "$MODE" in
...
         (((f.action // "") == "fix-pr" and has_url(f.fix_pr_url)) or has_url(f.suppress_pr_url)) as $needs_ta_medic
-        | "• " + icon(f) + " *" + catlabel(f) + "* — `"
-        + s(f.spec // f.operationId; "?") + "` — " + s(f.test; "")
-        + "\n    expected: " + cap(f.expected; "?"; 120) + "  |  actual: " + cap(f.actual; "?"; 120)
-        + (if (f.known_issue // false) then link_or_note(f.known_issue_url; ":ticket: known issue") else "" end)
-        + (if (f.related_commit // null) != null then "\n    :fast_forward: skipped — explained by recent change: " + s(f.related_commit; "") else "" end)
-        + (if (f.issue_url // null) != null then link_or_note(f.issue_url; ":memo: filed") else "" end)
+        | ([
+            (if (f.known_issue // false) then compactLink(f.known_issue_url; ":ticket:") else "" end),
+            (if (f.related_commit // null) != null then ":fast_forward: recent change" else "" end),
```
[the remark this answers is not in the document]
Comment (esraagamal6) [reply scripts/triage/hub-triage-format-slack.sh]: Fixed — `related_commit` now renders as an actual link when it is a real http(s) URL, or as inline text when it is a bare sha, instead of collapsing to a generic marker. See 3ba0870.

File: scripts/triage/hub-triage-format-slack.sh [hunk 2]
```diff
@@ -203,21 +219,34 @@ case "$MODE" in
       # top-level summary message) so it stays precise, not a blanket ping.
       def hub_medic: "<!subteam^S014VK4482H|hub-medic>";
       def test_automation_medic: "<!subteam^S09UF0EV0HG|test-automation-medic>";
+      # An admission from the agent that it could not confidently pick a
+      # category — never silently folded into an ordinary infrastructure
+      # finding. Called out with its own line AND routed to the
+      # test-automation medic so a human makes the call, rather than the
+      # low confidence going unnoticed under a category label that reads
+      # as a settled decision.
+      def undecided(f): (f.confidence // "") == "low";
```
[the remark this answers is not in the document]
Comment (esraagamal6) [reply scripts/triage/hub-triage-format-slack.sh]: Fixed — narrowed `undecided()` to the documented exact fallback combination (`category == "infrastructure" AND confidence == "low"`), not `confidence == "low"` alone. See 3ba0870.

File: scripts/triage/hub-triage-format-slack.sh [hunk 3]
```diff
@@ -203,21 +219,34 @@ case "$MODE" in
...
+        ((((f.action // "") == "fix-pr" and has_url(f.fix_pr_url)) or has_url(f.suppress_pr_url)) or undecided(f)) as $needs_ta_medic
+        | ([
+            (if (f.known_issue // false) then compactLink(f.known_issue_url; ":ticket:") else "" end),
+            (if (f.related_commit // null) != null then ":fast_forward: recent change" else "" end),
+            compactLink(f.issue_url; ":memo:"),
+            compactLink(f.fix_pr_url; if (f.action // "") == "skip" then ":recycle:" else ":hammer_and_wrench:" end),
+            compactLink(f.suppress_pr_url; ":no_entry:")
+          ] | map(select(length > 0)) | join("  ")) as $links_line
+        | "• " + icon(f) + " " + catlabel(f) + " — `" + s(f.operationId; "?")
+        + "` — expected " + leadNum(f.expected; "?") + ", got " + leadNum(f.actual; "?")
```
[the remark this answers is not in the document]
Comment (esraagamal6) [reply scripts/triage/hub-triage-format-slack.sh]: Fixed — added a spec fallback for a missing/empty operationId (via a new opLabel() helper; a plain `//` wouldn't have caught the empty-string case). See 3ba0870.

File: scripts/triage/hub-triage-format-slack.sh [hunk 4]
```diff
@@ -228,13 +257,11 @@ case "$MODE" in
...
-        "• :no_entry_sign: *unmapped* — `" + s(u.operationId; "?") + "` — no generated test"
-        + (if (u.fix_pr_url // null) != null then
-             link_or_note(u.fix_pr_url;
-               if (u.action // "") == "skip" then ":recycle: already being fixed" else ":hammer_and_wrench: fix PR" end)
-           else "" end)
-        + (if (u.action // "") == "fix-pr" and has_url(u.fix_pr_url)
-           then "\n    :rotating_light: " + test_automation_medic else "" end)
+        ((u.action // "") == "fix-pr" and has_url(u.fix_pr_url)) as $needs_ta_medic
+        | (compactLink(u.fix_pr_url; if (u.action // "") == "skip" then ":recycle:" else ":hammer_and_wrench:" end)) as $link
+        | "• :no_entry_sign: unmapped — `" + s(u.operationId; "?") + "` — no generated test"
```
[the remark this answers is not in the document]
Comment (esraagamal6) [reply scripts/triage/hub-triage-format-slack.sh:284]: Checked the schema — unmapped_operations entries only ever carry {operationId, action, fix_pr_url, file_error}, no spec field (unlike failures[], which does have one). There's nothing to fall back to here, and the pre-PR code never had this fallback either. Leaving uline() as-is.
